### PR TITLE
fix: move to next 14 🫶🏻

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

🥁  Next 14 upgrade 🥁  

Seems to works fine now :) 
Not sure if they themselves fixed the SVGR issue in the last release, but seems stable now.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://feat-next-14.preview.app.daily.dev

### CODE REVIEW

**Review by omBratteng — APPROVED** [review]
YOLO 🦾
(Change wise it looks good)

Should perhaps also test in preview

File: packages/shared/package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "graphql": "^15.5.0",
     "graphql-request": "^3.4.0",
     "idb-keyval": "^5.0.6",
-    "next": "^12.2.2",
+    "next": "14.2.8",
```
Comment (capJavert) [inline packages/shared/package.json:24]: Probably not just next but main bundle jumped by ~19kb https://vercel.com/dailydotdev/daily-webapp/8bvbNVp4a3bn1sG6wYfkv5W6nKpr#L157 vs main, something to maybe take a look at

Comment (capJavert) [reply packages/shared/package.json:24]: https://github.com/vercel/next.js/issues/58047

File: packages/webapp/next.config.js [hunk 2]
```diff
@@ -67,10 +84,6 @@ module.exports = withTM(
         CURRENT_VERSION: version,
       },
       rewrites: () => [
-        {
-          source: '/api/sitemaps/:path*',
-          destination: `${process.env.NEXT_PUBLIC_API_URL}/sitemaps/:path*`,
-        },
```
Comment (capJavert) [inline packages/webapp/next.config.js:73]: Removed accidentally?

**Review by capJavert — APPROVED** [review]
Changes wise nothing crucial, good work on this 💪

File: packages/webapp/next.config.js (hunk 2 above)
Comment (rebelchris) [reply packages/webapp/next.config.js:73]: Oh weird didn't need to indeed :O

**Review by sshanzel — APPROVED** [review]
Santa gave gifts early this year 😂 

Great work on the surprise upgrade 💪

File: packages/extension/package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "graphql": "^15.7.2",
     "graphql-request": "^3.6.1",
     "idb-keyval": "^5.1.5",
-    "next": "^12.2.2",
+    "next": "14.2.8",
```
Comment (omBratteng) [inline packages/extension/package.json:32]:
```suggestion
    "next": "14.2.9",
```
New version just dropped

**Review by omBratteng — CHANGES_REQUESTED** [review]

**Review by nensidosari — APPROVED** [review]
Looks great!! Tested in preview as well and couldn't find any issue.

**Review by omBratteng — APPROVED** [review]

**Pull request merged**